elastic/kibana
# [Content management] Add "Last updated" metadata to TableListView

In this PR I've updated the implementation of the `<TableListView />` component (exposed by the `kibana_react` plugin) to render a "Last updated" column if the items of the table contain the `updatedAt` metadata.

I have then updated the consuming apps to include the `updatedAt` from their saved objects when fetching the table items.

## Note on implementation

I've decided to render relative date ("a minute ago", "yesterday", "2 days ago") for up to 7 days ago. If an object has been updated more than 7 days ago we then render this format: `"May 17, 2022"`

| Less than 7 days  | More than 7 days |
| ------------- | ------------- |
| <img width="206" alt="Screenshot 2022-05-18 at 15 31 45" src="https://user-images.githubusercontent.com/2854616/169051032-17ccce41-5a6c-4f87-a4e9-6f7c01569825.png"> | <img width="213" alt="Screenshot 2022-05-18 at 15 32 19" src="https://user-images.githubusercontent.com/2854616/169051073-372b2e33-2e5b-45ee-8a2c-ed993501136f.png"> |


When there is no data for a specific saved object then a dash `"-"` is displayed with a tooltip

<img width="225" alt="Screenshot 2022-05-18 at 14 56 17" src="https://user-images.githubusercontent.com/2854616/169046768-5752baca-072e-42b9-ba96-e461bffbf9af.png">


## How to test

* Launch Kibana and load sample data
* Navigate to the different list views (Dashboard, visualization, maps, graph) and verify that there is a "Last updated" column.
* Verify that the default sorting is "DESC" on the "Last updated" column
* Edit one of the object and save it. Navigate back to the listing to verify that the updated at is indeed updated

# Release note

The list view for Dashboards, Visualizations, Maps and Graph has a new "Last updated" column to easily access content that has been recently modified.

## Updated apps

<img width="1194" alt="Screenshot 2022-05-17 at 15 22 18" src="https://user-images.githubusercontent.com/2854616/168823386-90d4f9ea-9055-4622-a78d-c5431e0a25e4.png">

<img width="1202" alt="Screenshot 2022-05-17 at 15 22 03" src="https://user-images.githubusercontent.com/2854616/168823413-c6befcdb-60a1-43eb-9b5c-a06d4249f2a9.png">

<img width="1241" alt="Screenshot 2022-05-17 at 15 21 42" src="https://user-images.githubusercontent.com/2854616/168823450-0820cd92-fff3-43fc-874a-00d5dd2f79ce.png">

<img width="1219" alt="Screenshot 2022-05-17 at 15 21 20" src="https://user-images.githubusercontent.com/2854616/168823468-419523fe-52a2-45ab-8161-b933fe10a3ec.png">


Fixes https://github.com/elastic/kibana/issues/132134
